spark/src/main/scala/io/delta/sql/DeltaSparkSessionExtension.scala

```diff
@@ -107,6 +107,11 @@ class DeltaSparkSessionExtension extends (SparkSessionExtensions => Unit) {
     extensions.injectPostHocResolutionRule { session =>
       new PreprocessTableDelete(session.sessionState.conf)
     }
+    // Resolve new UpCast expressions that might have been introduced by [[PreprocessTableUpdate]]
+    // and [[PreprocessTableMerge]].
```

Comment:
@johanl-db, in DeltaMergeBuilder we explicitly call PostHocResolveUpCast. The inject Rule only runs automatically for UPDATES and not for MERGE?

Reply:
MERGE using the Scala Delta table API doesn't go through the analyzer but instead resolves the merge command manually.  UPDATE (SQL + Scala/Java/Python) and MERGE (SQL) do run through the regular path and injected rules will be applied.

I'm not very familiar with why we need to resolve the MERGE command manually when using the Scala API, this may be something we can improve in the future